qt/python/mantidqt/widgets/superplot/view.py

```diff
@@ -0,0 +1,476 @@
+
+        for i in range(ws_item.childCount()):
+            sp_item = ws_item.child(i)
+            if sp_item.get_spectrum_index() == spectrum_index:
+                brush = sp_item.foreground(0)
+                brush.setColor(QColor(color))
+                sp_item.setForeground(0, brush)
+                sp_item.setText(0, label)
+
+    def set_workspaces_list(self, names):
+        """
+        Set the list of selected workspaces and update the workspace slider
+        length.
+
+        Args:
+            names (list(str)): list if the workspace names
+        """
+        self._side_view.workspacesList.blockSignals(True)
+        self._side_view.workspacesList.clear()
+        for name in names:
+            item = WorkspaceItem(self._side_view.workspacesList, name)
+            item.signals.sig_del_clicked.connect(
+                    self._presenter.on_del_button_clicked)
+            item.setText(0, name)
+        self._side_view.workspacesList.blockSignals(False)
+
+    def set_spectra_list(self, name, nums):
+        """
+        Set the list of spectrum index for a workspace in the list.
+
+        Args:
+            name (str): name of the workspace
+            nums (list(int)): list of the spectrum indexes
+        """
+        self._side_view.workspacesList.blockSignals(True)
+        ws_item = self._side_view.workspacesList.findItems(
+                name, Qt.MatchExactly, 0)[0]
+        ws_item.takeChildren()
+        for num in nums:
+            item = SpectrumItem(ws_item, num)
+            item.signals.sig_del_clicked.connect(
+                    self._presenter.on_del_spectrum_button_clicked)
+            item.setText(0, str(num))
+        self._side_view.workspacesList.expandAll()
+        self._side_view.workspacesList.blockSignals(False)
+
+    def get_spectra_list(self, name):
+        """
+        Get the list of spectrum indexes for a workspace in the list.
+
+        Args:
+            name (str): name of the workspace
+
+        Returns:
+            list(int): list of the spectrum indexes
+        """
+        ws_item = self._side_view.workspacesList.findItems(name,
+                                                           Qt.MatchExactly,
+                                                           0)
+        if ws_item:
+            ws_item = ws_item[0]
+        else:
+            return []
+        nums = list()
+        for i in range(ws_item.childCount()):
+            nums.append(ws_item.child(i).get_spectrum_index())
+        return nums
+
+    def check_hold_button(self, state):
+        """
+        Set the check state of the hold button. This function does not trigger
+        the button signals.
+
+        Args:
+            state (bool): check state
+        """
+        self._bottom_view.holdButton.blockSignals(True)
+        self._bottom_view.holdButton.setChecked(state)
+        self._bottom_view.holdButton.blockSignals(False)
+
+    def set_spectrum_disabled(self, state):
```

Comment:
Consider renaming for ease of understanding
```suggestion
    def set_spectrum_slider_disabled(self, state):
```

Reply:
I changed to `set_spectrum_selection_disabled`